runtime/src/accounts_db.rs

```diff
@@ -2133,16 +2163,21 @@ impl AccountsDb {
             ancestors,
             range,
             |pubkey, (account_info, slot)| {
+                // unlike other scan fns, this is called from Bank::collect_rent_eagerly(),
+                // which is on-consensus processing in the banking/replaying stage.
+                // This requires infallible and consistent account loading.
+                // So, we unwrap Option<LoadedAccount> from get_loaded_account() here.
+                // This is safe because this closure is invoked with the account_info,
+                // while we lock the index entry at AccountsIndex::do_scan_accounts() ultimately,
+                // meaning no other subsystems can invalidate the account_info before making their
+                // changes to the index entry.
+                // For details, see the comment in retry_to_get_account_accessor()
```

Comment:
this is another additional subtlety I found while finishing up this pr... pretty hard to spot...

Reply:
ha yup, this one is tricky. Lucky there's no bug in existing code for this case thought!